### PR TITLE
Add retrieve-neat-setting function

### PR DESCRIPTION
**Pull request by whmii**
`retrieve-neat-setting` takes the default map $_neat-grid-default and merges it
with the map passed in to it (typically `$_neat-grid`) and then retrieves
the value passed into it.

### CODE REVIEW

File: core/neat/functions/_retrieve-neat-settings.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/// @argument {string} $setting
+///
+/// @return {boolean | color | list | number | string}
+///
+/// @example scss
+///   _retrieve-neat-setting(columns)
+///
+/// @access private
+
+@function _retrieve-neat-setting($_grid, $setting) {
```
Comment (tysongach) [inline core/neat/functions/_retrieve-neat-settings.scss]: Can we update the name of the partial to match the name of the function? The file name currently has an `s` at the end.

File: core/neat/settings/_settings.scss [hunk 2]
```diff
@@ -11,7 +11,19 @@
 ///
 /// @access private
 
-$neat-default-grid: (
+$_neat-grid-default: (
```
Comment (tysongach) [inline core/neat/settings/_settings.scss]: Should this be `-defaults` (with an “s”) since it houses multiple default settings?

Comment (tysongach) [inline core/neat/settings/_settings.scss]: That would also go in line with the `$_bourbon-defaults` map.

**Comment by tysongach** [comment]
Couple minor notes, otherwise 👍

**Comment by whmii** [comment]
You know what… I did this wrong… damn

**Comment by tysongach** [comment]
@whmii How so?

**Comment by whmii** [comment]
The mixins should default to pulling in the `$neat-grid` not `$_neat-default-grid`

Also the function documentation is not correct.

**Comment by whmii** [comment]
To clary the approach here. it is the job of the `_retrieve-neat-setting()` function to merge the map passed into it with the `$_neat-grid-defaults` and return the value for the key it was given. It is then the job of the mixins to default to giving the function the `$neat-grid`

Just a bit of a seporation of labor here but one that is worth noting

**Comment by tysongach** [comment]
We have the same need in Bourbon: https://github.com/thoughtbot/bourbon/blob/master/core/bourbon/utilities/_retrieve-bourbon-setting.scss

Can we just pull that over here?

File: core/neat/settings/_settings.scss [hunk 3]
```diff
@@ -11,7 +11,19 @@
 ///
 /// @access private
 
-$neat-default-grid: (
+$_neat-grid-defaults: (
   columns: 12,
   gutter: 20px
 );
+
+/// User overrides of Bourbon configuration settings.
```
Comment (tysongach) [inline core/neat/settings/_settings.scss]: @whmii This description needs to be updated for Neat, not Bourbon.

**Comment by tysongach** [comment]
@whmii I’m confused. You indicated above that there were errors, but then this got merged any way. I’m not sure I understand your comment above.

Also, in the future, can you add details to how things work in the Git commit message, as well? It really helps out in the future when we’re trying to figure out why we did something the way we did.

**Pull request merged**